server/build_event_protocol/build_event_handler/build_event_handler.go

```diff
@@ -733,38 +733,38 @@ func (e *EventChannel) isFirstStartedEvent(bazelBuildEvent *build_event_stream.B
 	if e.hasReceivedStartedEvent {
 		return false
 	}
-	_, ok := bazelBuildEvent.Payload.(*build_event_stream.BuildEvent_Started)
+	_, ok := bazelBuildEvent.GetPayload().(*build_event_stream.BuildEvent_Started)
 	return ok
 }
 
 func (e *EventChannel) isFirstEventWithOptions(bazelBuildEvent *build_event_stream.BuildEvent) bool {
-	switch p := bazelBuildEvent.Payload.(type) {
+	switch p := bazelBuildEvent.GetPayload().(type) {
 	case *build_event_stream.BuildEvent_Started:
-		return p.Started.OptionsDescription != "" && !e.hasReceivedEventWithOptions
+		return p.Started.GetOptionsDescription() != "" && !e.hasReceivedEventWithOptions
 	case *build_event_stream.BuildEvent_OptionsParsed:
 		return !e.hasReceivedEventWithOptions
 	}
 	return false
 }
 
 func isWorkspaceStatusEvent(bazelBuildEvent *build_event_stream.BuildEvent) bool {
-	switch bazelBuildEvent.Payload.(type) {
+	switch bazelBuildEvent.GetPayload().(type) {
```

Comment:
Do these type-checks do the right thing for what I assume is an untyped field? Probably, but can you just confirm?

Reply:
`bazelBuildEvent.GetPayload()` is `bazelBuildEvent.Payload` with a nil check wrapper

```go
func (x *BuildEvent) GetPayload() isBuildEvent_Payload {
	if x != nil {
		return x.Payload
	}
	return nil
}
```

so the only difference is when `bazelBuildEvent` is nil. That should not happen as we initialize the empty struct on line 976.

But if it turns nil in a future iteration of the code, then we should still be safe and not panic. Example: https://go.dev/play/p/TPTsK5C7PlX